### DISCUSSION TITLE
A module to install .rpm files

### DISCUSSION THREAD

**Discussion by marcinjahn** [blue-build/.github#25] [Support] [answered]
Some packages are distributed as `.rpm` files without any copr repo or anything like that.
What is the best way to install such packages?

**Comment by fiftydinar** [comment]
You can use a script to place/download the file in `/tmp/myrpm/name-of-package.rpm` f.e. & than issue `rpm-ostree install /tmp/myrpm/name-of-package.rpm`

But I agree that it would be nice to have this integrated inside `rpm-ostree` module.

**Comment by xynydev** [comment, marked as answer]
The `rpm-ostree` module might already support installing RPMs straight from a URL, as I believe the `rpm-ostree install` command does too.

If that is the case, the issue comes from the module using `rpm-ostree override remove --install` when the `remove` section is present, which might not support the feature. 

You should try this by adding the URL to the RPM in the install section of an `rpm-ostree` module without a remove section. If that works, we shall make the module detect URLs to install them properly with `rpm-ostree install`, and document the feature.

**Comment by marcinjahn** [comment]
You're right, using an URL as a package seems to work fine: https://github.com/marcinjahn/mj-os/actions/runs/9032211800/job/24819959513
